src/main/scala/no/ndla/imageapi/service/ImageConverter.scala

```diff
@@ -24,48 +41,89 @@ trait ImageConverter {
         override def stream = new ByteArrayInputStream(outputStream.toByteArray)
         override def contentType: String = originalImage.contentType
         override def fileName: String = originalImage.fileName
+        override lazy val sourceImage = ImageIO.read(stream)
       }
     }
 
     def resize(originalImage: ImageStream, targetWidth: Int, targetHeight: Int): Try[ImageStream] = {
-      val sourceImage = ImageIO.read(originalImage.stream)
+      val sourceImage = originalImage.sourceImage
       Try(Scalr.resize(sourceImage, min(targetWidth, sourceImage.getWidth), min(targetHeight, sourceImage.getHeight)))
         .map(resized => toImageSteam(resized, originalImage))
     }
 
     def resize(originalImage: ImageStream, mode: Mode, targetSize: Int): Try[ImageStream] = {
-      val sourceImage = ImageIO.read(originalImage.stream)
-      val maxSize = mode match {
-        case Mode.FIT_TO_WIDTH => sourceImage.getWidth
-        case Mode.FIT_TO_HEIGHT => sourceImage.getHeight
-        case _ => max(sourceImage.getWidth, sourceImage.getHeight)
-      }
-      Try(Scalr.resize(sourceImage, mode, min(targetSize, maxSize))).map(resized => toImageSteam(resized, originalImage))
+      val MaxTargetSize = 2000
+      val sourceImage = originalImage.sourceImage
+      Try(Scalr.resize(sourceImage, mode, min(targetSize, MaxTargetSize))).map(resized => toImageSteam(resized, originalImage))
     }
 
     def resizeWidth(originalImage: ImageStream, size: Int): Try[ImageStream] = resize(originalImage, Mode.FIT_TO_WIDTH, size)
 
     def resizeHeight(originalImage: ImageStream, size: Int): Try[ImageStream] = resize(originalImage, Mode.FIT_TO_HEIGHT, size)
 
-    def crop(originalImage: ImageStream, start: Point, end: Point): Try[ImageStream] = {
-      val sourceImage = ImageIO.read(originalImage.stream)
-      val (topLeft, bottomRight) = transformCoordinates(start, end)
+    private def crop(image: ImageStream, sourceImage: BufferedImage, topLeft: PixelPoint, bottomRight: PixelPoint): Try[ImageStream] = {
       val (width, height) = getWidthHeight(topLeft, bottomRight, sourceImage)
 
       Try(Scalr.crop(sourceImage, topLeft.x, topLeft.y, width, height))
-        .map(cropped => toImageSteam(cropped, originalImage))
+        .map(cropped => toImageSteam(cropped, image))
+    }
+
+    def crop(originalImage: ImageStream, start: PercentPoint, end: PercentPoint): Try[ImageStream] = {
+      val sourceImage = originalImage.sourceImage
+      val (topLeft, bottomRight) = transformCoordinates(sourceImage, start, end)
+      crop(originalImage, sourceImage, topLeft, bottomRight)
+    }
+
+    private def getStartEnd(size: Int, targetSize: Int, focal: Int): (Int, Int) = {
```

Comment:
`getStartEnd` er et veldig generelt funksjonsnavn til å bare handle om fokuspunkt-varianten. Se forøvrig kommentaren nedenfor ang. gruppering av funksjoner.